dataset_api.md

```diff
@@ -0,0 +1,158 @@
+Enhancements:
+P5. Support passing of iterables of Terms to triples, quads, and related methods, similar to the triples_choices method.
+P6. Default the triples method to iterate with `(None, None, None)`
+
+With all of the above changes, including those changes proposed by Martynas, here are some examples:
+
+```python
+from rdflib import Dataset, Graph, URIRef, Literal
+from rdflib.namespace import RDFS
+
+# ============================================
+# Adding Data to the Dataset
+# ============================================
+
+# Initialize the dataset
+d = Dataset()
+
+# Add a single triple to the Default Graph, and a single triple to a Named Graph
+g1 = Graph()
+g1.add(
+    (
+        URIRef("http://example.com/subject-a"),
+        URIRef("http://example.com/predicate-a"),
+        Literal("Triple A")
+    )
+)
+# merge with the default graph
+d.default_graph += g1
+# or set the default graph
+d.default_graph = g1
+
+# Add a Graph to a Named Graph in the Dataset.
+g2 = Graph()
+g2.add(
+    (
+        URIRef("http://example.com/subject-b"),
+        URIRef("http://example.com/predicate-b"),
+        Literal("Triple B")
+    )
+)
+d.add_named_graph(name=URIRef("http://example.com/graph-B"), g2)
+
+# ============================================
+# Iterate over the entire Dataset returning triples
+# ============================================
+
+for triple in d.triples():
+    print(triple)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-a'), rdflib.term.URIRef('http://example.com/predicate-a'), rdflib.term.Literal('Triple A'))
+(rdflib.term.URIRef('http://example.com/subject-b'), rdflib.term.URIRef('http://example.com/predicate-b'), rdflib.term.Literal('Triple B'))
+
+# ============================================
+# Iterate over the entire Dataset returning quads
+# ============================================
+
+for quad in d.quads():
+    print(quad)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-a'), rdflib.term.URIRef('http://example.com/predicate-a'), rdflib.term.Literal('Triple A'), None)
+(rdflib.term.URIRef('http://example.com/subject-b'), rdflib.term.URIRef('http://example.com/predicate-b'), rdflib.term.Literal('Triple B'), rdflib.term.URIRef('http://example.com/graph-B'))
+
+# ============================================
+# Get the Default graph
+# ============================================
+
+dg = d.default_graph  # same as current default_context
+
+# ============================================
+# Iterate on triples in the Default Graph only
+# ============================================
+
+for triple in d.triples(graph="default"):
+    print(triple)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-a'), rdflib.term.URIRef('http://example.com/predicate-a'), rdflib.term.Literal('Triple A'))
+
+# ============================================
+# Access quads in Named Graphs only
+# ============================================
+
+for quad in d.quads(graph="named"):
```

Comment:
Shouldn't this be equivalent to simply `d.quads()`? Since the default graph does not produce quads.

Reply:
Yes the proposal is to have the "graph" of triples in the default graph set to None.